### DISCUSSION TITLE
I purged an OSD before checking the cluster is clean

### DISCUSSION THREAD

**Discussion by budimanjojo** [rook/rook#9576] [Q&A] [answered]
Hello, I have a 3 nodes cluster and one of the mon just broke itself. So I deleted the osd on the mon, `rm -rf /var/lib/rook` in that node and format the disk clean (I scaled down the `rook-ceph-operator` to 0 before doing this). After that I scale up the operator pod and hoping rook can create the new osd for me, but it just gives me this in the log:
```
2022-01-14 08:11:13.642951 I | clusterdisruption-controller: osd "rook-ceph-osd-2" is down but no node drain is detected
2022-01-14 08:11:14.412422 I | clusterdisruption-controller: osd is down in failure domain "budimanjojo-kmaster1" and pgs are not active+clean. pg health: "cluster is not fully clean. PGs: [{StateName:active+undersized Count:51} {StateName:active+undersized+degraded Count:46}]"
```
Is there anyway to recover from this? Or should I just start over again? Thank you

**Comment by travisn** [comment, marked as answer]
@budimanjojo 
- Were you trying to clean the whole node? 
- Did the mon on that node start up again successfully? For example, does `ceph status` show the mons are all three in quorum?
- If you really want to get rid of an OSD, see [these instructions](https://rook.github.io/docs/rook/v1.8/ceph-osd-mgmt.html#remove-an-osd), otherwise ceph will always look for that osd
- The operator should create a new OSD if that device really is cleaned and available. If it seems stuck on the OSD or PGs error, you might set [skipUpgradeChecks: false](https://github.com/rook/rook/blob/master/deploy/examples/cluster.yaml#L37) and try restarting the operator again.

**Reply by budimanjojo** [reply]
>     * Were you trying to clean the whole node?
> 
No, I just want to clean the OSD of that node
>     * Did the mon on that node start up again successfully? For example, does `ceph status` show the mons are all three in quorum?
> 
Yes it did, the mon pod is running and in quorum, except the mon used to be `mon_a` and because I deleted `/var/lib/rook` directory the new mon become `mon_d`.
>     * If you really want to get rid of an OSD, see [these instructions](https://rook.github.io/docs/rook/v1.8/ceph-osd-mgmt.html#remove-an-osd), otherwise ceph will always look for that osd
> 
Yes I followed the guide except I didn't confirm the PG status to be clean before proceeding to purge the OSD (because the cluster was already in degraded state since something bad happened to that drive).
>     * The operator should create a new OSD if that device really is cleaned and available. If it seems stuck on the OSD or PGs error, you might set [skipUpgradeChecks: false](https://github.com/rook/rook/blob/master/deploy/examples/cluster.yaml#L37) and try restarting the operator again.
>
Maybe that's what I didn't try before. But actually I already nuked the whole cluster and restored the data from backup already. So I think I will just close this discussion for now.

Thanks for the answer. :)
